ospfd/ospf_ism.h

```diff
@@ -53,8 +53,9 @@
 			listnode_add((O)->oi_write_q, oi);                     \
 			oi->on_write_q = 1;                                    \
 		}                                                              \
-		thread_add_write(master, ospf_write, (O), (O)->fd,             \
-				 &(O)->t_write);                               \
+		if (!list_isempty((O)->oi_write_q))                            \
```

Comment:
Based on your comment, I don't think this fully fixes the problem.

There are two asserts - first, that the list is not empty (`listhead()` is nonnull), second, that the data pointer of the list head is nonnull. This check works around the situation in which `oi->on_write_q == 1`, but it's not actually on the list - in which case we turn on the write thread, and when it runs, the first assert will be hit. But that is a buggy condition that we should be checking for anyway, if that is what is occurring.

Now if we are crashing on the second `assert(oi)`, where `oi` is fetched as the data pointer from the list node, this check doesn't fix that because we are just checking that at least one list node is present - not that its data pointer is nonnull.

Reply:
after a discussion w/ quentin he is right.  I've gone back to the drawing board and believe I found the issue.  pr updated